### PR TITLE
Add a custom font loader

### PR DESCRIPTION
**Pull request by tsauerwein**
This PR proposes to add a custom font loader. By overriding the file `mapfish-spring-custom-fonts.xml` in applications, custom fonts embedded in the web-application can be loaded.

```
    <bean id="custom-font-loader" class="org.mapfish.print.config.CustomFontLoader">
        <constructor-arg type="java.util.Set" index="0">
            <util:set>
                <value>fonts/ComingSoon.ttf</value>
            </util:set>
        </constructor-arg>
    </bean>
```

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/config/CustomFontLoader.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public CustomFontLoader(final Set<String> paths) {
+        this.paths = paths;
+    }
+
+    /**
+     * Load the custom fonts when the application is started.
+     */
+    @PostConstruct
+    public void init() {
+        for (String path : this.paths) {
```
Comment (pvalsecc) [inline core/src/main/java/org/mapfish/print/config/CustomFontLoader.java]: Why don't you do the loading in the constructor? I thought you wanted to do all that ASAP?

File: core/src/main/resources/mapfish-spring-application-context.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <import resource="mapfish-spring-config-output-formats.xml" />
     <import resource="mapfish-spring-access-assertions.xml" />
     <import resource="mapfish-spring-named-styles.xml" />
+    <import resource="mapfish-spring-custom-fonts.xml" />
```
Comment (pvalsecc) [inline core/src/main/resources/mapfish-spring-application-context.xml]: Again, why not move that at the top of the list of imported files?

File: core/src/main/java/org/mapfish/print/config/CustomFontLoader.java (hunk 1 above)
Comment (tsauerwein) [inline core/src/main/java/org/mapfish/print/config/CustomFontLoader.java]: This JasperReports font cache is only created when you actually generate a report. But yes, it is probably not a bad idea to register the fonts as soon as possible.

**Comment by tsauerwein** [comment]
PR updated, thanks for the review!

**Comment by pvalsecc** [comment]
ship it!

**Pull request merged**